gamemodes/terrortown/entities/entities/ttt_game_text.lua

```diff
@@ -54,28 +54,23 @@ end
 function ENT:AcceptInput(name, activator)
 	if name == "Display" then
 		local recv = activator
-
-		local r = self.Receiver
-		if r == RECEIVE_ALL then
-			recv = nil
-		elseif r == RECEIVE_DETECTIVE then
-			recv = GetRoleChatFilter(ROLE_DETECTIVE)
-		elseif r == RECEIVE_TRAITOR then
-			recv = GetTeamChatFilter(TEAM_TRAITOR)
-		elseif r == RECEIVE_INNOCENT then
-			recv = GetTeamChatFilter(TEAM_INNOCENT)
-		elseif r == RECEIVE_ACTIVATOR then
-			if not IsValid(activator) or not activator:IsPlayer() then
-				ErrorNoHalt("ttt_game_text tried to show message to invalid !activator\n")
-
-				return true
-			end
-		elseif r == RECEIVE_CUSTOMROLE and self.teamReceiver then
-			recv = GetTeamChatFilter(self.teamReceiver)
+				
+		if IsValid(activator) and activator:IsPlayer() then
+			local receiver_tbl = { -- i think this is faster than chaining if/elseif
+				RECEIVE_ALL = nil,
+				RECEIVE_DETECTIVE = GetRoleChatFilter(ROLE_DETECTIVE),
+				RECEIVE_TRAITOR = GetRoleChatFilter(TEAM_TRAITOR),
+				RECEIVE_INNOCENT = GetRoleChatFilter(TEAM_INNOCENT)
+			}
+			
+			recv = (self.teamReceiver) ? GetTeamChatFilter(self.teamReceiver) : receiver_tbl[self.Receiver]
+			CustomMsg(recv, self.Message, self.Color)
+			return true -- success
+		else
+			ErrorNoHalt("ttt_game_text tried to show message to invalid !activator\n")
+			return false -- failed, either invalid activator or activator was not a player
 		end
-
-		CustomMsg(recv, self.Message, self.Color)
-
-		return true
 	end
+	
+	return false -- failed, name wasn't "Display"
```

Comment:
Same as above: why the changed return value?

Reply:
it just seemed more logically correct to return a fail if the operation did not succeed with the provided input

as it stands, returning True really only tells us that name did = "Display"
if it returns False, then it tells us that either the name wasn't "Display", or the Activator was invalid/not a player

although, now that ive written this out... it actually does make more sense troubleshooting-wise to return True... that accompanied with the ErrorNoHalt tells us exactly what happened...

although, ErrorNoHalt has the same effect regardless of return value

**tl;dr imo ErrorNoHalt means we dont have to sacrifice what i believe is logical consistency**
it makes more sense to me that we'd return False if the operation failed, and True if it succeeded